src/sentry/static/sentry/app/views/projectDetail/projectTeamAccess.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {Organization, Project} from 'app/types';
+import {SectionHeading} from 'app/components/charts/styles';
+import {t, tn} from 'app/locale';
+import Link from 'app/components/links/link';
+import space from 'app/styles/space';
+import Placeholder from 'app/components/placeholder';
+import IdBadge from 'app/components/idBadge';
+import Button from 'app/components/button';
+
+type Props = {
+  organization: Organization;
+  project?: Project | null;
+};
+
+type State = {
+  collapsed: boolean;
+};
+
+class ProjectTeamAccess extends React.Component<Props, State> {
+  state: State = {
+    collapsed: true,
+  };
+
+  static MAX_WHEN_COLLAPSED = 5;
+
+  onCollapseToggle = () => {
+    this.setState(prevState => ({
+      collapsed: !prevState.collapsed,
+    }));
+  };
+
+  renderInnerBody() {
+    const {project, organization} = this.props;
+
+    if (!project) {
+      return <Placeholder />;
+    }
+
+    if (project.teams.length === 0) {
+      const hasPermission = organization.access.includes('project:write');
+      return (
+        <Button
+          to={`/settings/${organization.slug}/projects/${project.slug}/teams/`}
+          disabled={!hasPermission}
+          title={hasPermission ? undefined : t('You do not have permission to do this')}
+          priority="primary"
+        >
+          {t('Assign Team')}
+        </Button>
+      );
+    }
+
+    const {teams} = project;
+    const {collapsed} = this.state;
+
+    const canExpand = teams.length > ProjectTeamAccess.MAX_WHEN_COLLAPSED;
+
+    let teamsToRender = teams;
+    if (collapsed && canExpand) {
+      teamsToRender = teams.slice(0, ProjectTeamAccess.MAX_WHEN_COLLAPSED);
+    }
+
+    const numberOfCollapsedTeams = teams.length - teamsToRender.length;
```

Comment:
If we end up with a few more of these collapsible sections we should consider making a more abstract component for them.

Reply:
This is something that's on my radar.
There will be no more collapsible sections on this page, but this functionality is copy-pasted from the release detail sidebar and I bet it would be useful to have something like this in the future too. Basically, like our clippedBox component that does not take pixel height but a number of items (and can be collapsed back!).